### ISSUE TITLE
Please add aws.cloudfront.getDistributions()

### ISSUE THREAD

**Issue by Rez0k** [pulumi/pulumi-aws#3397]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Please add aws.cloudfront.getDistributions() to get list of available cloudfront distributions

### Affected area/feature

Cloudfront, docs

**Comment by t0yv0** [comment]
Thank you for the enhancement request! 

If I understand correctly this style of querying is not available in the upstream TF provider, correct? Looking at 

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/cloudfront_distribution

I see a datasource that can get one cloudfront distribution but not query all of them. I would be easiest for Pulumi provider to adopt this feature if it was available upstream, so perhaps it could be helpful to open a ticket there (I could not find one at the moment)?

**Comment by Rez0k** [comment]
> Thank you for the enhancement request!
> 
> If I understand correctly this style of querying is not available in the upstream TF provider, correct? Looking at
> 
> https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/cloudfront_distribution
> 
> I see a datasource that can get one cloudfront distribution but not query all of them. I would be easiest for Pulumi provider to adopt this feature if it was available upstream, so perhaps it could be helpful to open a ticket there (I could not find one at the moment)?

Yes, I could not find it at terraform, I tried to use `cloudfront.getDistribution` (https://www.pulumi.com/registry/packages/aws/api-docs/cloudfront/getdistribution/) to solve my problem, but I saw that the function requires an id in order to work, but the function also accept tags.
I can't get the ID in my case, but I should be able to get the cloudfront distribution by tags.
I don't understand why ID is required, why can't I just get the distribution by tag without specifying ID.
Is it a bug?

if not, how can I get cloufront distribution using tags only if possible?
